internal/controller/prefectserver_controller.go

```diff
@@ -363,9 +389,9 @@ func pvcNeedsUpdate(current, desired *corev1.PersistentVolumeClaimSpec, log logr
 	return false
 }
 
-func jobNeedsUpdate(current, desired *batchv1.JobSpec, log logr.Logger) bool {
-	// TODO: check for changes to the job spec that require an update
-	return false
+func migrationJobNeedsUpdate(current, desired *batchv1.JobSpec, log logr.Logger) bool {
+	merged := current.DeepCopy()
+	return needsUpdate(current, merged, desired, log)
```

Comment:
originally i thought to stash the prefect version / postgres connection info in a configmap and do an update check on that resource to determine if the migration job should be re-created

but we store that info in the Job spec already, in the image (which is computed from the CR version) and the env vars (where we pass in the CR's postgres connection parameters).   

im thinking we just do the same deepmerge check on the Job spec itself since it contains all of the version/db connection information, and if any of that changes we can recreate the Job